File: ios/OTPPoc/OTPPoc/Views/PhoneInputView.swift
import SwiftUI

struct PhoneInputView: View {
    @ObservedObject var viewModel: AuthViewModel

    var body: some View {
        VStack(spacing: AppSpacing.innerGap) {
            VStack(alignment: .leading, spacing: 8) {
                Text("Enter your phone number")
                    .font(.system(size: 18, weight: .bold))
                    .foregroundColor(AppColors.textPrimary)

                Text("We'll send a one-time code to verify your identity.")
                    .font(.system(size: 13))
                    .foregroundColor(AppColors.textSecondary)
            }
            .frame(maxWidth: .infinity, alignment: .leading)

            HStack(spacing: 10) {
                Image(systemName: "phone.fill")
                    .foregroundColor(AppColors.textMuted)
                    .font(.system(size: 15))

                TextField("+1 555 123 4567", text: $viewModel.phone)
                    .keyboardType(.phonePad)
                    .textContentType(.telephoneNumber)
                    .font(.system(size: 17))
                    .onChange(of: viewModel.phone) {
                        viewModel.recordInteraction()
                    }
            }
            .padding(14)
            .background(AppColors.surface)
            .cornerRadius(AppSpacing.inputRadius)
            .overlay(
                RoundedRectangle(cornerRadius: AppSpacing.inputRadius)
                    .stroke(AppColors.border, lineWidth: 1)
            )

            Button {
                Task { await viewModel.startAuth() }
            } label: {
                Text("Send OTP")
                    .font(.system(size: 16, weight: .semibold))
                    .frame(maxWidth: .infinity)
                    .frame(height: AppSpacing.buttonHeight)
                    .background(viewModel.phone.isEmpty ? AppColors.textMuted : AppColors.brand500)
                    .foregroundColor(.white)
                    .cornerRadius(AppSpacing.buttonRadius)
            }
            .disabled(viewModel.phone.isEmpty)

            if case .error(let message) = viewModel.state {
                HStack(spacing: 10) {
                    Image(systemName: "exclamationmark.triangle.fill")
                        .foregroundColor(AppColors.error)
                        .font(.system(size: 14))
                    Text(message)
                        .font(.system(size: 13))
                        .foregroundColor(AppColors.error)
                        .frame(maxWidth: .infinity, alignment: .leading)
                }
                .padding(12)
                .background(AppColors.errorBg)
                .cornerRadius(AppSpacing.inputRadius)
            }
        }
        .padding(AppSpacing.cardPadding)
        .background(AppColors.surface)
        .cornerRadius(AppSpacing.cardRadius)
        .shadow(color: .black.opacity(0.06), radius: 12, x: 0, y: 4)
        .padding(.horizontal, 20)
    }
}

File: android/app/src/main/java/com/otppoc/android/views/PhoneInputView.kt
package com.otppoc.android.views

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Phone
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.otppoc.android.ui.theme.AppColors
import com.otppoc.android.ui.theme.AppSpacing
import com.otppoc.android.viewmodels.AuthState
import com.otppoc.android.viewmodels.AuthViewModel

@Composable
fun PhoneInputView(viewModel: AuthViewModel) {
    val state by viewModel.state.collectAsState()
    val phone by viewModel.phone.collectAsState()

    Column(
        modifier = Modifier
            .padding(horizontal = 20.dp)
            .shadow(
                elevation = 4.dp,
                shape = RoundedCornerShape(AppSpacing.cardRadius),
                ambientColor = Color.Black.copy(alpha = 0.06f)
            )
            .background(AppColors.surface, RoundedCornerShape(AppSpacing.cardRadius))
            .padding(AppSpacing.cardPadding),
        verticalArrangement = Arrangement.spacedBy(AppSpacing.innerGap)
    ) {
        Column(
            modifier = Modifier.fillMaxWidth(),
            verticalArrangement = Arrangement.spacedBy(8.dp)
        ) {
            Text(
                text = "Enter your phone number",
                fontSize = 18.sp,
                fontWeight = FontWeight.Bold,
                color = AppColors.textPrimary
            )
            Text(
                text = "We'll send a one-time code to verify your identity.",
                fontSize = 13.sp,
                color = AppColors.textSecondary
            )
        }

        OutlinedTextField(
            value = phone,
            onValueChange = {
                viewModel.updatePhone(it)
                viewModel.recordInteraction()
            },
            placeholder = {
                Text("+1 555 123 4567", color = AppColors.textMuted)
            },
            leadingIcon = {
                Icon(
                    imageVector = Icons.Filled.Phone,
                    contentDescription = null,
                    tint = AppColors.textMuted,
                    modifier = Modifier.size(20.dp)
                )
            },
            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Phone),
            singleLine = true,
            shape = RoundedCornerShape(AppSpacing.inputRadius),
            colors = OutlinedTextFieldDefaults.colors(
                focusedBorderColor = AppColors.brand500,
                unfocusedBorderColor = AppColors.border,
                focusedContainerColor = AppColors.surface,
                unfocusedContainerColor = AppColors.surface
            ),
            modifier = Modifier.fillMaxWidth()
        )

        Button(
            onClick = { viewModel.startAuth() },
            enabled = phone.isNotEmpty(),
            shape = RoundedCornerShape(AppSpacing.buttonRadius),
            colors = ButtonDefaults.buttonColors(
                containerColor = AppColors.brand500,
                disabledContainerColor = AppColors.textMuted
            ),
            modifier = Modifier
                .fillMaxWidth()
                .height(AppSpacing.buttonHeight)
        ) {
            Text(
                text = "Send OTP",
                fontSize = 16.sp,
                fontWeight = FontWeight.SemiBold,
                color = Color.White
            )
        }

        // Error banner
        if (state is AuthState.Error) {
            val errorMessage = (state as AuthState.Error).message
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .background(
                        AppColors.errorBg,
                        RoundedCornerShape(AppSpacing.inputRadius)
                    )
                    .padding(12.dp),
                horizontalArrangement = Arrangement.spacedBy(10.dp),
                verticalAlignment = Alignment.Top
            ) {
                Icon(
                    imageVector = Icons.Filled.Warning,
                    contentDescription = null,
                    tint = AppColors.error,
                    modifier = Modifier.size(16.dp)
                )
                Text(
                    text = errorMessage,
                    fontSize = 13.sp,
                    color = AppColors.error,
                    modifier = Modifier.weight(1f)
                )
            }
        }
    }
}
